vars.tf

```diff
@@ -1,52 +1,80 @@
-variable "aws_region" {
+variable "acm_certificate_arn" {
   type        = string
-  description = "AWS Region"
+  description = "(optional) ARN of certificate for load balancer to use. Defaults to auto-creating a new certificate"
+  default     = null
 }
 
-variable "ecr_keycloak_image_url" {
+variable "aws_region" {
   type        = string
-  description = "ECR Keycloak Image location"
+  description = "AWS Region"
 }
 
-variable "log-bucket-owner-id" {
+variable "database_subnet_group" {
   type        = string
-  description = "IAM id for the logs bucket owner (differs per region)"
+  description = "(optional) Name of a database subnet group where Keycloak MariaDB instance will reside. Defaults to creating a new subnet group containing var.private_subnets"
+  default     = null
 }
 
-variable "public_subnets" {
-  description = "List of public subnets"
+variable "db_name" {
+  type        = string
+  description = "Keycloak database name"
 }
 
-variable "private_subnets" {
-  description = "List of private subnets"
+variable "db_username" {
+  type        = string
+  description = "Keycloak database username"
 }
 
-variable "availability_zones" {
-  description = "List of availability zones"
+variable "ecr_keycloak_image_url" {
+  type        = string
+  description = "(optional) ECR Keycloak Image location. Defaults to creating a new ECR repo"
+  default     = null
```

Comment:
Where are we getting our Keycloak image from? Should we default to that.

Reply:
The module uses the `null` value to determine whether to create a new ECR repo, so that has to be the default.

Reply:
@ianwestcott Where are we getting our ECS image though? Are we creating it ourselves, or downloading something from Docker Hub, such as [this](https://registry.hub.docker.com/r/jboss/keycloak#!)? I'm not too familiar with ECS structure, so sorry if I'm merging concepts here. If we are creating a plug and play module for others to use, then defaulting to the docker image url would make sense. And then the comparison would look something like this: `count = var.ecr_keycloak_image_url == ...local.keycloak_image_url.default... ? 1 : 0`

Reply:
@grejdi-mbta that's actually an open question that we're still working out with the vendor, but our hope is that we'll be building a Docker image from a Dockerfile in a separate MBTA repo. If there's a publicly-available fallback (e.g. the stock Keycloak image available in Docker Hub), I'd rather document that in the README as part of some "how to use this module" documentation, which I plan to add in the future. In either case, uploading the Docker image to ECR would happen outside of Terraform.